src/utils.ts

```diff
@@ -142,7 +218,7 @@ const getSmartWalletAddress = async ({
 
   const recovererAddress = recoverer ?? constants.AddressZero;
 
-  const logicParamsHash = data ?? SHA3_NULL_S;
+  const logicParamsHash = data ? utils.keccak256(data) : SHA3_NULL_S;
```

Comment:
Are we sure that `data` isn't already hashed?

Reply:
It could be that is already hashed but it will because the user is sending the data in a wrong way.
As you can see [here](https://github.com/rsksmart/rif-relay-contracts/blob/main/contracts/factory/CustomSmartWalletFactory.sol#L205), the data is hashed but [here](https://github.com/rsksmart/rif-relay-contracts/blob/main/contracts/factory/CustomSmartWalletFactory.sol#L250), is not, so unless that we change the contract we need to hash it.